### DISCUSSION TITLE
Generate pre-signed URLs?

### DISCUSSION THREAD

**Discussion by thedodd** [Byron/google-apis-rs#275] [Q&A] [answered]
Pretty impressive project here. I am looking for a convenience function for generating pre-signed URLs for access to objects in storage.

Any ideas on that?

**Comment by Byron** [comment]
For reference, here is the [documentation on pre-signed urls](https://cloud.google.com/storage/docs/access-control/signed-urls).

Despite me having never used this feature and just having looked it up, I believe you should be able to use the app-identity API to do this. However, I didn’t find it in the [API-list](http://byron.github.io/google-apis-rs/) of this project, so getting it might be a manual call away.

Also I don’t know how they are actually used. I hope that answers your question.

**Comment by mwilliammyers** [comment, marked as answer]
Not sure if you or anyone is still looking for this but you are probably looking for the `*sign_blob` methods in the [iam](https://docs.rs/google-iam1/1.0.8+20181005/google_iam1/struct.ProjectServiceAccountSignBlobCall.html) crate or the [iamcredentials](https://docs.rs/google-iamcredentials1/1.0.8+20181004/google_iamcredentials1/struct.ProjectServiceAccountSignBlobCall.html) crate. I haven't used either crate  and I am not even sure what the difference between the two is, but I have used the corresponding methods in other API libraries to do what you want.

**Reply by mwilliammyers** [reply]
See also: https://github.com/google-apis-rs/generator/issues/21 and [this playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c1ca71c3da61d47f6f14caa5fb107124)
